src/ci/resources/slurm_application.conf

```
@@ -9,13 +9,29 @@ backend {
       config {
         runtime-attributes = """
           String? docker
-          String? docker_user
         """
 
         # https://slurm.schedmd.com/sbatch.html
         submit-docker = """
-        sbatch -J ${job_name} -D ${cwd} --wrap \
-          "/bin/bash -c 'docker run --rm ${"--user " + docker_user} --entrypoint ${job_shell} -v ${cwd}:${docker_cwd} ${docker} ${docker_script}'"
+            set -euo pipefail
+            CACHE_DIR=$HOME/.singularity/cache
+            mkdir -p $CACHE_DIR
+            LOCK_FILE=$CACHE_DIR/singularity_pull_flock
+            DOCKER_NAME=$(sed -e 's/[^A-Za-z0-9._-]/_/g' <<< ${docker})
+            IMAGE=$CACHE_DIR/$DOCKER_NAME.sif
+            (
+              flock --verbose --exclusive --timeout 900 9
```

genuinely not sure this is a good idea for this use case but throwing it out there 😅 

```suggestion
              flock --verbose --exclusive --timeout 900 9 || exit 1
```